### PR TITLE
idk if this works, but here's a small clip implementation

### PR DESCRIPTION
**Pull request by korakoe**

### CODE REVIEW

**Comment by korakoe** [comment]
I have to redownload my dataset so idk if it works

**Comment by korakoe** [comment]
THIS IS VERY BROKEN AND DOES NOT WORK BECAUSE IM NOT SMART ENOUGH

(works for training, not for inference, if someone wants to take a look at my poorly written code then why not)

**Comment by korakoe** [comment]
The main reason this doesnt work is because idk how to do negative embeddings here, nor do I know how to filter the logits from clip
